sympy/simplify/hyperexpand_doc.py

```diff
@@ -1,18 +1,90 @@
+        'Ei': ':func:`~sympy.functions.special.error_functions.Ei`',
+        'Ci': ':func:`~sympy.functions.special.error_functions.Ci`',
+        'Si': ':func:`~sympy.functions.special.error_functions.Si`',
+        'Shi': ':func:`~sympy.functions.special.error_functions.Shi`',
+        'Chi': ':func:`~sympy.functions.special.error_functions.Chi`',
+        'fresnels': ':func:`~sympy.functions.special.error_functions.fresnels`',
+        'fresnelc': ':func:`~sympy.functions.special.error_functions.fresnelc`',
 
-__doc__ = doc
+        # Elliptic integrals
+        'elliptic_k': ':func:`~sympy.functions.special.elliptic_integrals.elliptic_k`',
+        'elliptic_e': ':func:`~sympy.functions.special.elliptic_integrals.elliptic_e`',
+
+        # Elementary functions - exponential
+        'exp': ':func:`~sympy.functions.elementary.exponential.exp`',
+        'log': ':func:`~sympy.functions.elementary.exponential.log`',
+
+        # miscellaneous
+        'sqrt': ':func:`~sympy.functions.elementary.miscellaneous.sqrt`',
+        'root': ':func:`~sympy.functions.elementary.miscellaneous.root`',
+
+        # trigonometric
+        'cos': ':func:`~sympy.functions.elementary.trigonometric.cos`',
+        'sin': ':func:`~sympy.functions.elementary.trigonometric.sin`',
+
+        # hyperbolic
+        'cosh': ':func:`~sympy.functions.elementary.hyperbolic.cosh`',
+        'sinh': ':func:`~sympy.functions.elementary.hyperbolic.sinh`',
+
+        # Combinatorial functions
+        'RisingFactorial': ':func:`~sympy.functions.combinatorial.factorials.RisingFactorial`',
+        'factorial': ':func:`~sympy.functions.combinatorial.factorials.factorial`',
+
+        # Other special functions
+        'lerchphi': ':func:`~sympy.functions.special.zeta_functions.lerchphi`',
+
+    }
+
+    def get_special_functions_used(expr):
+        """Extract special function names from an expression"""
+        func_names = set()
+        for atom in expr.atoms(Function):
+            if not isinstance(atom, AppliedUndef):
+                func_name = atom.func.__name__
+                if func_name in FUNCTION_LINKS:
+                    func_names.add(func_name)
+        return sorted(func_names)
+
+    c = FormulaCollection()
+
+    doc = ""
+
+    for f in c.formulae:
+        obj = Eq(hyper(f.func.ap, f.func.bq, f.z),
+                f.closed_form.rewrite('nonrepsmall'))
+        doc += ".. math::\n  %s\n" % latex(obj)
+        # Add links to special functions used
+        special_funcs = get_special_functions_used(f.closed_form)
+        if special_funcs:
+            doc += "\n"
+            if len(special_funcs) == 1:
+                doc += "This formula uses %s.\n" % FUNCTION_LINKS[special_funcs[0]]
+            else:
+                func_links = ', '.join(FUNCTION_LINKS[fn] for fn in special_funcs[:-1])
+                doc += "This formula uses %s and %s.\n" % (func_links, FUNCTION_LINKS[special_funcs[-1]])
```

Comment:
This assumes that the number of functions is exactly 2 but without checking that. I think that the code should be written a way that would raise an error in that case if there are not currently any examples where there are more than 2 functions.

Rather than building doc as a string with repeated `+=` it is better to make a list of strings and join them with `doc = '\n'.join(lines)` at the end.